File: source/go/bst_inorder.go
package main

import (
	"fmt"
	"strings"
)

type Node struct {
	value int
	left  *Node
	right *Node
}

func insert(root *Node, value int) *Node {
	if root == nil {
		return &Node{value: value}
	}
	if value < root.value {
		root.left = insert(root.left, value)
	} else if value > root.value {
		root.right = insert(root.right, value)
	}
	return root
}

func inorder(root *Node, result *[]int) {
	if root == nil {
		return
	}
	inorder(root.left, result)
	*result = append(*result, root.value)
	inorder(root.right, result)
}

func main() {
	var root *Node
	values := []int{50, 30, 70, 20, 40, 60, 80}
	for _, v := range values {
		root = insert(root, v)
	}

	var result []int
	inorder(root, &result)
	parts := make([]string, len(result))
	for i, v := range result {
		parts[i] = fmt.Sprintf("%d", v)
	}
	fmt.Printf("inorder: %s\n", strings.Join(parts, " "))
}

File: source/javascript/bst_inorder.js
// BST in-order traversal
function newNode(value) { return { value, left: null, right: null }; }

function insert(root, value) {
    if (root === null) return newNode(value);
    if (value < root.value) root.left = insert(root.left, value);
    else if (value > root.value) root.right = insert(root.right, value);
    return root;
}

function inorder(root, result) {
    if (root === null) return;
    inorder(root.left, result);
    result.push(root.value);
    inorder(root.right, result);
}

let root = null;
const values = [50, 30, 70, 20, 40, 60, 80];
for (const v of values) root = insert(root, v);

const result = [];
inorder(root, result);
console.log(`inorder: ${result.join(" ")}`);
